### ISSUE TITLE
Incorrect handling of deleted thread root messages 

### ISSUE THREAD

**Issue by janogarcia** [vector-im/element-web#21466]
### Steps to reproduce

1. Make sure you have `Settings > Preferences > Timeline > Show a placeholder for removed messages` **disabled**.
2. Create a thread.
3. Delete the root message.

**Note**: We decided long ago to keep thread root messages always visible, even if the `Show a placeholder for removed messages` is disabled.

### Outcome

It seems a regression has been introduced for deleted root messages, breaking the previous correct behavior.

Now, deleting a root message will:

- 🔴 Make the root message and thread summary disappear completely from the main timeline.
- 🔴 Keep showing the original contents for the deleted message in the thread timeline.
- ✅ Show the correct  "Message deleted" status in the thread list.

What should happen instead:

Please refer to the [Figma mockups](https://www.figma.com/file/T309ztx0sNyOOK6NKVLHsK/Threads?node-id=1150%3A202558) for the detailed context/flows.

In short:

- Show the "Message deleted" state for the root message and keep displaying its thread summary in the main timeline (instead of _"Make the root message and thread summary disappear completely from the main timeline."_)
- Immediately stop showing the original content for the deleted message in the thread timeline (instead of _Keep showing the original contents for the deleted message in the thread timeline._)-

### URL for webapp

develop.element.io

### Application version

Element version: 071a5410b88e-react-4e4ce65f584b-js-a3f5ec1ba2fa Olm version: 3.2.8

### Will you send logs?

Yes

**Comment by t3chguy** [comment]
> 🔴 Make the root message and thread summary disappear completely from the main timeline.

So I think the issue is a bit more nuanced, the thread is still there, attached to the `Message deleted`, its just that its collapsed into the Summary. @janogarcia what should we do here?

![image](https://user-images.githubusercontent.com/2403652/159483691-bd908e8f-6c00-4e82-8644-b019fa59fad6.png)

**Comment by janogarcia** [comment]
@t3chguy Unsure if I've understood your remarks.

I've edited the issue description to hopefully add more context. Please find all the details about the expected behavior in the Figma mockups for the [Deleting the root message Flow](https://www.figma.com/file/T309ztx0sNyOOK6NKVLHsK/Threads?node-id=1150%3A202558).

It had been working correctly for months I'd say, only recently stopped working (last week) .

Maybe @gsouquet can clarify / shed some light on your question?

**Comment by janogarcia** [comment]
@t3chguy Another odd behavior that was introduced las week on deleted messages:

As stated in the issue description, the root message contents are not immediately hidden from the thread timeline. It keeps showing the original message. When reloading the page the deleted root message will be rendered as expected in the thread timeline ("Deleted message" state) **but** preceded by a "You can't see earlier messages" notice which shouldn't be there.

![Element  4  | 🧪 Playground Encrypted](https://user-images.githubusercontent.com/60790/159521398-144e4d3b-ea49-436b-9999-c7bae0af2b62.png)

**Comment by janogarcia** [comment]
@t3chguy I was not aware of #21434, thanks. It happens consistently for the case I described, though, if that helps track it down.

**Comment by janogarcia** [comment]
@t3chguy Updated the issue description after further internal discussion. I added a note on the decision we made long ago about displaying deleted root messages even when the `Settings > Preferences > Timeline > Show a placeholder for removed messages` option is disabled.

**Comment by t3chguy** [comment]
Cannot reproduce even with that option disabled.

![image](https://user-images.githubusercontent.com/2403652/159679192-e5f09242-26f7-47c2-b8cd-6b94d3c0a780.png)
![image](https://user-images.githubusercontent.com/2403652/159679203-755699d9-5bb0-4ff7-b24e-86f24ed12953.png)

**Comment by janogarcia** [comment]
On my end, the deleted root messages consistently disappear from the main timeline as soon as I disable said option.

Element version: e638abf7c253-react-9b2944294cf0-js-d0b964837f28
Olm version: 3.2.8
